### PR TITLE
[Java] Eager operations implementation

### PR DESCRIPTION
**Pull request by karllessard**
Pull request #4/5 for enabling eager execution support in the Java client.

This one is implementing the `Operation` interface for eager environments, via `EagerOperation`, and include some additional unit tests.

CC: @sjamesr

### CODE REVIEW

**Review by sjamesr — DISMISSED** [review]

File: tensorflow/java/src/test/java/org/tensorflow/EagerOperationBuilderTest.java [hunk 1]
```diff
@@ -48,5 +48,92 @@ public void failToBuildOpIfSessionIsClosed() {
...
+  public void addInputs() {
+    try (EagerSession session = EagerSession.create()) {
+      Operation asrt = opBuilder(session, "Assert", "assert")
+          .addInput(TestUtil.constant(session, "Cond", true))
+          .addInputList(new Output<?>[] {TestUtil.constant(session, "Error", -1)})
+          .build();
+      try {
+        opBuilder(session, "Const", "var").addControlInput(asrt);
+        fail();
+      } catch (UnsupportedOperationException e) {}
```
Comment (rthadur) [inline tensorflow/java/src/test/java/org/tensorflow/EagerOperationBuilderTest.java]: failing internal lint test , here is the error: Empty catch block. [EmptyCatchBlock]

File: tensorflow/java/src/test/java/org/tensorflow/EagerOperationTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+package org.tensorflow;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
```
Comment (rthadur) [inline tensorflow/java/src/test/java/org/tensorflow/EagerOperationTest.java]: internal lint error: Missing a Javadoc comment. [GoogleJavadocType]

File: tensorflow/java/src/test/java/org/tensorflow/EagerOperationTest.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+public class EagerOperationTest {
+  
+  @Test
+  public void failToCreateIfSessionIsClosed() {
+    EagerSession session = EagerSession.create();
+    session.close();
+    try {
+      new EagerOperation(session, 1L, new long[] {1L}, "Add", "add");
+      fail();
+    } catch (IllegalStateException e) {}
```
Comment (rthadur) [inline tensorflow/java/src/test/java/org/tensorflow/EagerOperationTest.java]: lint error: Empty catch block. [EmptyCatchBlock]

File: tensorflow/java/src/test/java/org/tensorflow/EagerOperationTest.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+      try {
+        split.inputListLength("no_such_input");
+        fail();
+      } catch (IllegalArgumentException e) {}
+
+      try {
+        split.outputListLength("no_such_output");
+        fail();
+      } catch (IllegalArgumentException e) {}
```
Comment (rthadur) [inline tensorflow/java/src/test/java/org/tensorflow/EagerOperationTest.java]: lint error: Empty catch block. [EmptyCatchBlock]

File: tensorflow/java/src/test/java/org/tensorflow/EagerOperationTest.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+          .addInput(c1)
+          .setAttr("num_split", 2)
+          .build();
+      assertEquals(1, split.inputListLength("split_dim"));
+      assertEquals(2, split.outputListLength("output"));
+
+      try {
+        split.inputListLength("no_such_input");
+        fail();
+      } catch (IllegalArgumentException e) {}
```
Comment (rthadur) [inline tensorflow/java/src/test/java/org/tensorflow/EagerOperationTest.java]: lint error: Empty catch block. [EmptyCatchBlock]

File: tensorflow/java/src/test/java/org/tensorflow/EagerOperationTest.java [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    EagerOperation add = opBuilder(session, "Add", "SetDevice")
+        .addInput(TestUtil.constant(session, "Const1", 2))
+        .addInput(TestUtil.constant(session, "Const2", 4))
+        .build();
+    assertEquals(1, add.outputListLength("z"));
+    session.close();
+    try {
+      add.outputListLength("z");
+      fail();
+    } catch (IllegalStateException e) {}
```
Comment (rthadur) [inline tensorflow/java/src/test/java/org/tensorflow/EagerOperationTest.java]: lint error: Empty catch block. [EmptyCatchBlock]

**Comment by rthadur** [comment]
@karllessard there were some internal lint errors , can you please fix them

**Comment by karllessard** [comment]
@rthadur : fixed, thank you.

@sjamesr : do you need to reapprove? I was also looking to add those new lint checks to our list of [compiler rules](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/java/build_defs.bzl) but I did not find their equivalent, I guess those are internal rules only?

**Review by sjamesr — APPROVED** [review]

**Pull request merged**